packages/shared/src/guards.ts

```diff
@@ -27,8 +28,8 @@ export function assert(condition: unknown, message: string): asserts condition {
 
 export const isTokenSwap = (
   params: ExecuteSwapParams,
-): params is TokenSwapParams => 'srcAsset' in params;
+): params is TokenSwapParams => params.srcAsset !== Assets.ETH;
```

Comment:
im afraid this changes the bahaviour in the case where `srcAsset` is not set/undefined

Reply:
it should still be a token swap/call if `srcAsset` is not set